src/avram/needy_initializer.cr

```diff
@@ -0,0 +1,102 @@
+module Avram::NeedyInitializer
+  macro included
+    OPERATION_NEEDS = [] of Nil
+
+    macro inherited
+      inherit_needs
+    end
+  end
+
+  macro needs(type_declaration)
+    {% OPERATION_NEEDS << type_declaration %}
+    @{{ type_declaration.var }} : {{ type_declaration.type }}
+    property {{ type_declaration.var }}
+  end
+
+  macro inherit_needs
+    \{% if !@type.constant(:OPERATION_NEEDS) %}
+      OPERATION_NEEDS = [] of Nil
+    \{% end %}
```

Comment:
This doesn't seem necessary...

Reply:
🤷  lol. I know it's bad, but I literally copied and just ripped out code I assumed wasn't needed. I guess I should actually read what it is doing 😂

Reply:
I took a look at this again, and without it, things start to explode, so I'm leaving it in for now.